pysoa/server/logging.py

```diff
@@ -36,3 +38,151 @@ def get_logging_request_context(cls):
         if getattr(cls._logging_context, 'context_stack', None):
             return cls._logging_context.context_stack[-1]
         return None
+
+
+class RecursivelyCensoredDictWrapper(object):
+    # Some performance info about this set
+    #
+    # In [1]: %timeit 'password' in SENSITIVE_FIELDS
+    # The slowest run took 16.69 times longer than the fastest. This could mean that an intermediate result is being
+    # cached.
+    # 10000000 loops, best of 3: 129 ns per loop
+    #
+    # In [2]: %timeit 'nope' in SENSITIVE_FIELDS
+    # 10000000 loops, best of 3: 131 ns per loop
+    #
+    # In [3]: %timeit random_string()
+    # 100000 loops, best of 3: 10.4 us per loop
+    #
+    # In [4]: %timeit random_string() in SENSITIVE_FIELDS
+    # 100000 loops, best of 3: 10.7 us per loop
+    #
+    # Suffice it to say, a lookup against this set should always take less than 250 ns.
+    SENSITIVE_FIELDS = frozenset({
```

Comment:
I would add the auth token to this list.  Encrypted body content in general ought to be blanked.

Reply:
We discussed that in the squad channel on Slack. A request was made to //temporarily// exclude auth tokens from this while we work work out the kinks in our new internal authentication service, and then eventually add it to this list.